lib/os/cbprintf_complete.c

```diff
@@ -1327,6 +1348,32 @@ static int outs(cbprintf_cb out,
 	return (int)count;
 }
 
+/* Helper function for the cbvprintf() function below
+ *
+ * Update pad0 values based on precision and converted length.
+ */
+void update_pad0_values(struct conversion *const conv, int precision,
```

Comment:
This new code introduces a minor 10.3 violation ("Conversion from essential type to different or narrower essential type") in this function on line `:1363`

```
size_t len = bpe - bps; /* bpe and bps are chars */
```

It also introduces a minor 4.6 violation ("Using basic numerical type `int` rather than a typedef with size and signedness information") in the prototype on `:1355`:

```
void update_pad0_values(struct conversion *const conv, int precision, /* int */
```

I think that fixing one of the few violations of 15.3, which is Required, matters more than introducing a few violations, though. 10.3 is also Required but has thousands of violations already. 4.6 is Advisory, has thousands of violations and will probably be removed since many of our interfaces use basic types like `int`. I also think trying to fix those two violations will be more trouble than it's worth. I guess I'm just writing this comment just as a note for reference instead of a suggestion. :sweat_smile: